### PR TITLE
Rewritten queries to support stats plugin compatibility with PostgreSQL

### PR DESCRIPTION
**Pull request by bogdanJ9**
PostgreSQL support appeared in version 1.11 of SourceMod (development).

This pull request contains SQL script for creating tables in the database (misc\import_stats_pgsql.sql) and a plugin with query support.

### CODE REVIEW

**Comment by nickdnk** [comment]
@splewis You should probably merge my PR before this one, so this one can get updated to include the columns I added. Putting this one in first just means you'll have breaking changes twice.

**Comment by splewis** [comment]
Yes, could you add the new stats from #612:

* utility_damage 
* enemies_flashed 
* friendlies_flashed 
* knife_kills

File: scripting/get5_pgsqlstats.sp [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+      int firstdeath_ct = kv.GetNum(STAT_FIRSTDEATH_CT);
+      int tradekill = kv.GetNum(STAT_TRADEKILL);
+      int kast = kv.GetNum(STAT_KAST);
+      int contribution_score = kv.GetNum(STAT_CONTRIBUTION_SCORE);
+      int mvp = kv.GetNum(STAT_MVP);
+      
+      char teamString[16];
+      GetTeamString(team, teamString, sizeof(teamString));
+      
+      // TODO: this should really get split up somehow. Once it hits 32-arguments
```
Comment (nickdnk) [inline scripting/get5_pgsqlstats.sp:271]: Remove the 32 argument comment. It's no longer a limitation.

**Comment by nickdnk** [comment]
@bogdanJ9 Thanks for this. Please take a look at @splewis' comment and include all required columns. When you've done that and checked it works we can look into merging it.

File: scripting/get5_pgsqlstats.sp [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+  version = PLUGIN_VERSION,
+  url = "https://github.com/splewis/get5"
+};
+// clang-format on
+
+public void OnPluginStart() {
+  InitDebugLog("get5_debug", "get5_pgsql");
+  
+  g_ForceMatchIDCvar = CreateConVar(
+    "get5_pgsql_force_matchid", "0",
```
Comment (nickdnk) [inline scripting/get5_pgsqlstats.sp:55]: This needs to go and it should read the `matchid` instead. It is included in the forward that gets called when the series start. Look at the cases here: https://github.com/splewis/get5/pull/795/files and make sure that "scrim" and "manual" are both considered candidates for auto-increment.

File: scripting/get5_pgsqlstats.sp [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+  }
+}
+
+public void Get5_OnRoundStatsUpdated() {
+  if (Get5_GetGameState() == Get5State_Live && !g_DisableStats) {
+    UpdateRoundStats(MapNumber());
+  }
+}
+
+static int MapNumber() {
```
Comment (nickdnk) [inline scripting/get5_pgsqlstats.sp:339]: Remove this and use the native `Get5_GetMapNumber()`.

File: scripting/get5_pgsqlstats.sp [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+
+#include "get5/util.sp"
+
+#pragma semicolon 1
+#pragma newdecls required
+
+Database db = null;
+char queryBuffer[2048];
+
+int g_MatchID = -1;
```
Comment (nickdnk) [inline scripting/get5_pgsqlstats.sp:36]: Don't duplicate `g_MatchID` here. Use the `GetMatchId` method on the events from the forwards:

**Comment by nickdnk** [comment]
@bogdanJ9 Do you have any intention to complete this PR? We'll accept it if you update it. If not, we'll close it.

**Pull request closed without merging**